src/modules/groupAccess.ts

```diff
@@ -1,11 +1,36 @@
+import { prisma } from '../utils/prisma';
 import { AccessCheckResult, allowed, notAllowed } from '../utils/access';
 import { SessionUser } from '../utils/auth';
 
 import { tr } from './modules.i18n';
 
+const checkGroupAdmin = async (groupId: string, userId: string) => {
+    const admins = await prisma.$queryRaw<Array<{ count: number }>>`
+    SELECT count(*) FROM public."GroupAdmin"
+    WHERE "userId" = ${userId}
+    AND "groupId" in (
+        WITH RECURSIVE rectree AS (
+            SELECT *
+            FROM public."Group"
+            WHERE id = ${groupId}
+        UNION ALL
+            SELECT g.*
+            FROM public."Group" g
+            JOIN rectree
+                ON g.id = rectree."parentId"
+        ) SELECT id FROM rectree
+    )
+    `;
+    return !!Number(admins[0].count);
+};
+
 export const groupAccess = {
-    isEditable: (sessionUser: SessionUser): AccessCheckResult => {
-        if (sessionUser.role?.editFullGroupTree || sessionUser.role?.editAdministratedGroupTree) return allowed;
+    isEditable: async (sessionUser: SessionUser, groupId: string): Promise<AccessCheckResult> => {
+        if (sessionUser.role?.editFullGroupTree || sessionUser.role?.editAdministratedGroupTree) {
```

Comment:
wrong, we can allow editing without any checks only for those who can editFullTree

Reply:
done

Reply:
still wrong